website/guide/programming.html

```diff
@@ -493,6 +493,32 @@ <h2 id="duktape-api">Duktape API</h2>
 duk_get_prop_string(ctx, -1, "my_state_variable");
 </pre>
 
+
+<h3>Hidden Symbol properties</h3>
+
+<p>If data needs to be associated with an object, but hidden from Ecmascript
+code, hidden <a href="#symbols">Symbols</a> can be used as a property key. The
+key is only accessible via the C API, unless passed to Ecmascript code from the
+C API. A common use case is to associated backing pointers/data to C memory.
+Symbols are created as a string, but with a specific prefix.</p>
+
+<p>Setting then getting a hidden symbol on <code>this</code>.</p>
+<pre class="c-code">
+duk_push_this(ctx);
+duk_push_pointer(ctx, my_context_data);
+duk_put_prop_string(ctx, -2, "\xff" "\xff" "my_context_data");
+/* ... */
+duk_push_this(ctx);
+duk_get_prop_string(ctx, -1, "\xff" "\xff" "my_context_data");
+my_context_data_t *my_context_data = duk_get_pointer(ctx, -1);
+</pre>
+
+<p>See the <a href="#symbols">Symbols</a> section regarding the strange C
+literal string syntax.</p>
```

Comment:
This remark is IMO unnecessary with the API macro; it's not visible to calling code. It is still discussed in the Symbols section, and that is linked to already above.

Reply:
Sounds good.